scripts/taginfo.js

```diff
@@ -19,27 +52,86 @@ function addNetworkTags(project) {
       let network = entry[0],
         definition = entry[1];
 
-      let description = `Roads carrying routes in this network are marked by shields`;
+      let icon = definition.spriteBlank || definition.norefImage;
+      if (Array.isArray(icon)) {
+        icon = icon[0];
+      }
+
+      let icon_url;
+
+      if (icon == undefined) {
+        let shieldGfx = Shields.generateSpriteCtx({}, `shield\n${network}= `);
+        let def = JSON.stringify(shields[network]);
+
+        if (!shieldGfxMap.has(def)) {
+          shieldGfxMap.set(def, shieldGfxMap.size);
+        }
+
+        let network_filename_id = shieldGfxMap.get(def);
+        let save_filename = `dist/shield-sample/shield_${network_filename_id}.svg`;
+
+        if (!fs.existsSync(save_filename)) {
+          shieldGfx.canvas.saveAsSync(save_filename);
+        }
+        icon_url = `https://zelonewolf.github.io/openstreetmap-americana/shield-sample/shield_${network_filename_id}.svg`;
+      } else {
+        icon_url = `https://raw.githubusercontent.com/ZeLonewolf/openstreetmap-americana/main/icons/${icon}.svg`;
+      }
+
+      let description = `Roads carrying routes in this network are marked by `;
+      if (definition.canvasDrawnBlank) {
+        let shapeDef = definition.canvasDrawnBlank;
+        let prettyShapeName = `${shapeDef.drawFunc}`;
+        let prettyFillColor = getNamedColor(shapeDef.params.fillColor, "white");
+        let prettyStrokeColor = getNamedColor(
+          shapeDef.params.strokeColor,
+          "black"
+        );
+
+        switch (shapeDef.drawFunc) {
+          case "roundedRectangle":
+            if (shapeDef.params.radius == 10) {
+              prettyShapeName = "pill";
+            } else {
+              prettyShapeName = "rectangle";
+            }
+            break;
+          case "hexagonVertical":
+            prettyShapeName = "vertical hexagon";
+            break;
+          case "octagonVertical":
+            prettyShapeName = "vertical octagon";
+            break;
+          case "hexagonHorizontal":
+            prettyShapeName = "horizontal hexagon";
+            break;
+          case "ellipse":
+            if (shapeDef.params.rectWidth == 20) {
+              prettyShapeName = "circle";
+            }
+          case "pentagon":
+            if (shapeDef.params.angle == 0) {
+              prettyShapeName = "home plate";
+            }
+        }
+        description += `${prettyFillColor} ${prettyShapeName}-shaped shields with ${prettyStrokeColor} borders`;
+      } else {
+        description += "shields";
+      }
+
       if (definition.modifiers && definition.modifiers.length > 0) {
```

Comment:
If a programmatically generated shield has modifiers, do we need to output a separate sample image for it? For prefabricated images, we’re just using the base shield image, which tends to downscale better than a shield with stacked modifiers would:

<img src="https://user-images.githubusercontent.com/1231218/215983159-708d4ecb-9984-44a1-bb11-acf4d4ca593d.svg" height="16" alt="US:TX:Loop:Express">

Consider doing the same for the programmatic shields: delete `modifiers` before stringifying.

Reply:
I was thinking that we would _add_ modifiers to the pre-generated shield graphics, but I see now that that's a terrible idea.